client/app/bundles/shared/components/Dropdown/Dropdown.jsx

```diff
@@ -1,21 +1,33 @@
+// @flow
 import React from 'react';
 import shortid from 'shortid';
+import isNil from 'lodash/isNil';
+import { getAvailableLocales } from 'libs/i18n/I18nUtils';
+
 import css from './Dropdown.scss';
 
-import enYML from '../../../../../../config/locales/en.yml';
+const defaultLocales = getAvailableLocales();
 
+type Props = {
+    onChange: (e: string) => void,
+    locale: string,
+    localeList: { [key: string]: string },
+};
 
-const options = Object.keys(enYML.en.languages).map(key =>
-  (<option value={key} key={shortid.generate()} >
-    {enYML.en.languages[key]}
-  </option>),
-);
+export default (variationClassName: string) =>
+  ({ onChange, locale, localeList }: Props) => {
+    const localeOptions = isNil(localeList) ? defaultLocales : localeList;
 
-export default variationClassName => () => (
-  <div className={`${css.select_dropdown} ${variationClassName}`}>
-    <select>
-      {options}
-    </select>
-  </div>
-);
+    const options = Object.keys(localeOptions).map(key =>
+      (<option value={key} key={shortid.generate()} >
+        {localeOptions[key]}
+      </option>),
+    );
 
+    return (
+      <div className={`${css.select_dropdown} ${variationClassName}`}>
+        <select onChange={e => onChange(e.target.value)} value={locale || null}>
```

Comment:
Should value be null? Should we be only returning if locale exists?

Reply:
This can be made into an if statement. Null values are implicitly ignored but we can to ahead and make it explicit.

Reply:
Yeah let's make it explicit!